spec/image_spec.rb

```diff
@@ -26,200 +24,248 @@
+      let(:input_path) { File.join(TEST_INPUT_DIR, 'rgb.tif') }
+
       before do
-        generate_test_image(TEST_TIF_INPUT_FILE, compress: 'none', width: '37838', height: '37838')
+        generate_test_image(input_path, compress: 'none', width: '37838', height: '37838')
       end
 
       it 'creates the jp2 with a temp file' do
-        expect(File).to exist TEST_TIF_INPUT_FILE
-        expect(File).not_to exist TEST_JP2_OUTPUT_FILE
-        result = assembly_image.create_jp2(output: TEST_JP2_OUTPUT_FILE)
+        expect(File).to exist input_path
+        expect(File).not_to exist output_file
+        result = assembly_image.create_jp2(output: output_file)
         expect(assembly_image.tmp_path).not_to be_nil
         expect(result).to be_a_kind_of described_class
-        expect(result.path).to eq TEST_JP2_OUTPUT_FILE
-        expect(TEST_JP2_OUTPUT_FILE).to be_a_jp2
+        expect(result.path).to eq output_file
+        expect(output_file).to be_a_jp2
         expect(result.exif.colorspace).to eq 'sRGB'
-        jp2 = described_class.new(TEST_JP2_OUTPUT_FILE)
+        jp2 = described_class.new(output_file)
         expect(jp2.height).to eq 37_838
         expect(jp2.width).to eq 37_838
       end
     end
 
     context 'when given an LZW compressed RGB tif with more than 4GB of image data', skip: 'This test will create a 4GB temporary image, so skipping by default.' do
+      let(:input_path) { File.join(TEST_INPUT_DIR, 'lzw.tif') }
+
       before do
-        generate_test_image(TEST_TIF_INPUT_FILE, compress: 'lzw', width: '37838', height: '37838')
+        generate_test_image(input_path, compress: 'lzw', width: '37838', height: '37838')
       end
 
       it 'creates the jp2 with a temp file' do
-        expect(File).to exist TEST_TIF_INPUT_FILE
-        expect(File).not_to exist TEST_JP2_OUTPUT_FILE
-        result = assembly_image.create_jp2(output: TEST_JP2_OUTPUT_FILE)
+        expect(File).to exist input_path
+        expect(File).not_to exist output_file
+        expect(assembly_image.exif.samplesperpixel).to be 3
+        expect(assembly_image.exif.bitspersample).to eql '8 8 8'
+        expect(assembly_image).to be_a_valid_image
+        expect(assembly_image).to be_jp2abl
+        result = assembly_image.create_jp2(output: output_file)
         expect(assembly_image.tmp_path).not_to be_nil
         expect(result).to be_a_kind_of described_class
-        expect(result.path).to eq TEST_JP2_OUTPUT_FILE
-        expect(TEST_JP2_OUTPUT_FILE).to be_a_jp2
+        expect(result.path).to eq output_file
+        expect(output_file).to be_a_jp2
         expect(result.exif.colorspace).to eq 'sRGB'
-        jp2 = described_class.new(TEST_JP2_OUTPUT_FILE)
+        jp2 = described_class.new(output_file)
         expect(jp2.height).to eq 37_838
         expect(jp2.width).to eq 37_838
```

Comment:
@calavano double-checking -- what are the properties of the jp2 that we care about?  Size, colorspace and that it's a jp2?   Anything else?  samplesperpixel or bitspersample or whatever?  We're all image ignoramouses here.

Seems like we'd want to check these properties on every jp2 created by the code, no?